### DISCUSSION TITLE
About API

### DISCUSSION THREAD

**Discussion by mplatypus** [pear-devs/pear-desktop#2022] [Q&A]
Hi, I have noticed that there is an open PR about an API for the yt app! I am very excited for this. Would be cool to see it implemented soon.

My first question is, will there be a place to see what REST/WS methods to use, to access this API?

Second question is, I have been trying to implement my own API for The app (custom plugin), but its sorta hard, as I seem to be running into an issue where I can't really get anything to work (I can see my plugin, and all, but the backend doesn't work?) I am new to TS, (not really to JS) and my main language is python, but more of what I am asking, is that I can't seem to find much documentation for the creation of a plugin? its very lacking, for example, I can't seem to find any events towards tracks getting started/stopped, etc. Not sure if this is by design, but would be nice to get some clarity in this area.

**Comment by ArjixWasTaken** [comment]
There is no such event that I am aware of.
You could use the standard Media DOM Events to achieve that instead.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play_event
https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/pause_event

**Reply by mplatypus** [reply]
I am a bit curious of how the discord plugin does it then. I had a little look into it (actually took a fair bit of code from it) but even then, it doesn't quite seem to work.

**Reply by ArjixWasTaken** [reply]
> I am a bit curious of how the discord plugin does it then. I had a little look into it (actually took a fair bit of code from it) but even then, it doesn't quite seem to work.

The discord rich presence plugin doesn't need to run code whenever a track is paused/played.
It just pulls information at an interval.

**Reply by mplatypus** [reply]
oh that makes a lot of sense. Honestly, for my goal, that would work fine. Thank you for the insight!
